### ISSUE TITLE
Create a new file which just a copy of another exists file in the same folder is not possible.

### ISSUE THREAD

**Issue by zw963** [Alexander-Miller/treemacs#967]
Following screenshot is a example.

![image](https://user-images.githubusercontent.com/549126/182141386-d97dcce6-eff8-4943-953f-c10ca4ae9790.png)

I want create a `edit.ecr` on `companies` folder which is just a copy of `new.ecr`.

But, it seem like no way to do this, When i select `nf`(File new), it create a new empty file, when select `yf`(File copy),  treemacs create
a new folder `edit.ecr`, and copy `new.ecr` into it, like this:  `edit.ecr/new.ecr`(I use helm completion)

So, how to do that?

Thank you.

**Comment by Alexander-Miller** [comment]
>  When i select nf(File new), it create a new empty file

Pretty sure this isn't a default keybind. It should be `cf` for `create-file`. But you still get to pick the file's new name, right? If yes, that that part is working as intended.

> treemacs create a new folder edit.ecr, and copy new.ecr into it

This on the other hand sounds like a complete bug. Copying and moving had been changed to only let you pick the directory, without the option to rename. I did that because I had a common implementation for moving/copying a single file or multiple files at once (which I still need to document now that I think about it). 

However that is no longer the case. I've now changed the copy/file implementation to allow renaming. If you pick an existing directory as a target your file is moved/copied into that directory. If you pick something that isn't a directory then treemacs treats the target as a new name and will copy/move+rename your file.
